prog/victoria_metrics/victoria_metrics_server_nexus.rb

```diff
@@ -0,0 +1,172 @@
+      Strand.create(prog: "VictoriaMetrics::VictoriaMetricsServerNexus", label: "start") { _1.id = vs.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        hop_destroy
+      elsif strand.stack.count > 1
+        pop "operation is cancelled due to the destruction of the victoriametrics server"
+      end
+    end
+  end
+
+  label def start
+    nap 5 unless vm.strand.label == "wait"
+    victoria_metrics_server.incr_initial_provisioning
+
+    register_deadline("wait", 10 * 60)
+
+    cert, cert_key = create_certificate
+    victoria_metrics_server.update(cert: cert, cert_key: cert_key)
+
+    hop_bootstrap_rhizome
+  end
+
+  label def bootstrap_rhizome
+    bud Prog::BootstrapRhizome, {"target_folder" => "victoriametrics", "subject_id" => vm.id, "user" => "ubi"}
+
+    hop_wait_bootstrap_rhizome
+  end
+
+  label def wait_bootstrap_rhizome
+    reap
+    hop_create_victoriametrics_user if leaf?
+    donate
+  end
+
+  label def create_victoriametrics_user
+    begin
+      victoria_metrics_server.vm.sshable.cmd("sudo groupadd -f --system victoriametrics")
+      victoria_metrics_server.vm.sshable.cmd("sudo useradd --no-create-home --system -g victoriametrics victoriametrics")
+    rescue => ex
+      raise unless ex.message.include?("already exists")
+    end
+
+    hop_setup
+  end
+
+  label def setup
+    bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :install
+    bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :configure
+
+    hop_wait_setup
+  end
+
+  label def wait_setup
+    reap
+    if leaf?
+      hop_wait
+    end
+    donate
+  end
+
+  label def wait
+    when_reconfigure_set? do
+      bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :configure
+      hop_wait_reconfigure
+    end
+
+    when_restart_set? do
+      decr_restart
+
+      push self.class, frame, "restart"
+    end
+
+    if victoria_metrics_server.certificate_last_checked_at < Time.now - 60 * 60 * 24 * 30 # ~1 month
+      hop_refresh_certificates
+    end
+
+    nap 10
+  end
+
+  label def refresh_certificates
+    cert, cert_key = create_certificate
+    victoria_metrics_server.update(cert: cert, cert_key: cert_key, certificate_last_checked_at: Time.now)
+
+    incr_reconfigure
+    hop_wait
+  end
+
+  label def wait_reconfigure
+    decr_reconfigure
+
+    reap
+    if leaf?
+      hop_wait
+    end
+    donate
+  end
+
+  label def restart
+    case victoria_metrics_server.vm.sshable.cmd("common/bin/daemonizer --check restart_victoriametrics")
+    when "Succeeded"
+      victoria_metrics_server.vm.sshable.cmd("common/bin/daemonizer --clean restart_victoriametrics")
+      pop "victoriametrics server is restarted"
+    when "Failed", "NotStarted"
+      victoria_metrics_server.vm.sshable.cmd("common/bin/daemonizer 'systemctl restart victoriametrics && systemctl restart vmauth' restart_victoriametrics")
```

Comment:
consider daemonizer2, 496ebc92e9abfcdc17ecf3b1916221b255ac177e

Reply:
Nice, updated to use this. Looks shiny!